3rdparty/obiwarp/obiwarp.cpp

```diff
@@ -0,0 +1,136 @@
+#include "obiwarp.h"
+
+
+ObiWarp::ObiWarp(string score,bool local, float factor_diag, float factor_gap, float gap_init,float gap_extend,
+            float init_penalty, float response, bool nostdnrm ){
+
+    this->score = &score[0];
+    this->local = local;
+    this->factor_diag = factor_diag;
+    this->factor_gap = factor_gap;
+    this->gap_init = gap_init;
+    this->gap_extend = gap_extend;
+    this->init_penalty = init_penalty;
+    this->response = response;
+    this->nostdnrm = nostdnrm;
+    
+    tmPoint = NULL;
+    mzPoint = NULL;
+
+}
+ObiWarp::~ObiWarp(){
+    //release memory from all places
+    // if(tmPoint)
+    //     delete[] tmPoint;
+    // if(mzPoint)
+    //     delete[] mzPoint;
+}
+
+void ObiWarp::setReferenceData(vector<float> &rtPoints, vector<float> &mzPoints, vector<vector<float> >& intMat){
+    _tm_vals = rtPoints.size();
+    tmPoint = new float[_tm_vals];
+    for(int i=0; i < _tm_vals ; ++i)
+        tmPoint[i] = rtPoints[i];
+    _tm.take(_tm_vals, tmPoint);
+
+    _mz_vals = mzPoints.size();
+    mzPoint = new float[_mz_vals];
+    for(int i = 0; i < _mz_vals; ++i)
+        mzPoint[i] = mzPoints[i];
+    _mz.take(_mz_vals, mzPoint);
+
+    assert(_tm_vals == intMat.size());
+    MatF mat(_tm_vals, _mz_vals);
+    for(int i=0; i < _tm_vals; ++i){
+        assert(_mz_vals == intMat[i].size());
+        for(int j = 0; j < _mz_vals; ++j)
+            mat(i,j) = intMat[i][j];
+    }
+    _mat.take(mat);
+
+}
+
+vector<float> ObiWarp::align(vector<float> &rtPoints, vector<float> &mzPoints, vector<vector<float> >& intMat){
+    
+    vector<float> alignedRts;
+    if(rtPoints.size() == 0 || mzPoints.size() == 0 || _tm.len() == 0 || _mz.len() == 0)
+        return alignedRts;
```

Comment:
The function that takes `alignedRts`(i.e `Aligner::alignSampleRts`) should make sure to use the vector only if the size  of vector is not zero

Reply:
@rish9511 removed from here!